Sources/Target/Windows/Thread.cpp

```diff
@@ -33,6 +33,19 @@ Thread::Thread(Process *process, ThreadId tid, HANDLE handle)
 
 Thread::~Thread() { ::CloseHandle(_handle); }
 
+void Thread::setBreakpointStopReason() {
+  HardwareBreakpointManager *hwBpm = process()->hardwareBreakpointManager();
+  if (hwBpm != nullptr) {
+    BreakpointManager::Site site;
+    int bpIdx = hwBpm->hit(this, site);
+    if (bpIdx >= 0) {
+      setHardwareBreakpointStopReason(bpIdx, site);
+      return;
+    }
+  }
+  _stopInfo.reason = StopInfo::kReasonBreakpoint;
```

Comment:
Can't you hoist this and do an early return?

    _stopInfo.reason = StopInfo::kReasonBreakpoint;
    auto *manager = process()->hardwareBreakpointManager();
    if (manager == nullptr)
      return;

    BreakpointManager::Site site;
    int index = manager->hit(this, site);
    if (index < 0)
      return;

    setHardwareBreakpointStopReason(index, site);

Reply:
Mmm, good idea.